### DISCUSSION TITLE
Client side javascript Library for embedding charts and dashboards (similar to powerbi client library)

### DISCUSSION THREAD

**Discussion by sevakram** [apache/superset#19774] [Ideas]
Our client is evaluating Superset to phase out Powerbi. but a key requirement is embedding in superset in angular/react/vanilajs, where client generates a token from backend based on the current login. RLS and Authorization is decided based on token.

So the requirement is - A client side library for embedding superset using JavaScript or TypeScript into your apps. (Without iframe)

This is our reference
https://github.com/microsoft/PowerBI-JavaScript

Thanks

**Comment by chrisjung-dev** [comment]
You know that PowerBI embeds Charts and Stuff via IFRAMEs don't you? ;-)

**Comment by maxgillmaxgill** [comment]
Is your problem that it's an iFrame or are you trying to find a way to use your main app as an identity provider for Superset?
